pkg/spec/conflict.go

```diff
@@ -39,3 +40,42 @@ func createHeaderInConflictMsg(path *field.Path, in, in2 interface{}) string {
 func (c conflict) String() string {
 	return c.msg
 }
+
+// conflictSolver will get 2 types and returns
+// -1 - types conflict can't be resolved
+//
+//	0 - type1 and type2 are equal
+//	1 - type1 should be used
+//	2 - type2 should be used
+func conflictSolver(type1, type2 string) int {
+	if type1 == type2 {
+		return 0
+	}
+
+	if shouldPreferType(type1, type2) {
+		return 1
+	}
+
+	if shouldPreferType(type2, type1) {
+		return 2
+	}
+
+	return -1
+}
+
+// shouldPreferType return true if type1 should be preferred over type2
+// Note: MUST be called when type1 and type2 are not identical.
```

Comment:
why it MUST be called when type1 and type2 are not identical?

Reply:
`shouldPreferType` - id types are identical there is no need to prefer anything over anything :)
I don't want to add the identical check also in that function, I don't think it's part of the function role.

Reply:
I need `shouldPreferType` to first check if to prefer type 1 over type 2 and if not, I need to check if to prefer type 2 over type 1 - it is not the same check.

Reply:
And what will happen if you call this function with type 1 and type 2 identical? you will get true only if types are strings, otherwise false. don't think this is wrong. what do I care to prefer string over string? or not preferring any other types?

Reply:
ok, will remove the comment. it's a internal function anyway.